app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -166,21 +166,13 @@ def remote_feed_url
         uri.to_s
       end
 
-      def publication_options(repository_version)
```

Comment:
Musing: do you see any value in keeping this around as

```ruby
def publication_options(repository_version)
  {repository_version: repository_version}
end
```

for reasons like encapsulation, single-responsibility, readability, extensibility, etc?

In that case, we'd still keep `publication_data = api.publication_class.new(publication_options(href))` below (and could do the same for the verbatim, non-deb case, I suppose)

OR, in your experience is this a case of "You Aren't Gonna Need It"?

I don't have rigid, inflexible opinions about either approach, just curious whether you see any value in it based on your experience with this area of the code, & anticipating how it's likely to evolve in the future.

Reply:
I am going to go with: "We may never need it, so I lean towards cleaning it up". I had a look at the API docs for several Pulp plugins and there is the odd plugin specific option for Pulp distributions (some plugins don't have distributions), but the evidence suggests Katello never made use of any of those.